tests/test_outcome.py

```diff
@@ -203,17 +204,47 @@ def test_list_linked_outcomes(self, m):
                 ]
             }, m)
 
-        result = self.account_outcome_group.list_linked_outcomes()
+        with warnings.catch_warnings(record=True) as warning_list:
+            result = self.account_outcome_group.list_linked_outcomes()
+            self.assertIsInstance(result[0], OutcomeLink)
+            self.assertEqual(result[0].outcome_group['id'], 2)
+            self.assertEqual(result[0].outcome_group['title'], "Account Test Outcome Group")
+
+            result = self.canvas_outcome_group.list_linked_outcomes()
+            self.assertIsInstance(result[0], OutcomeLink)
+            self.assertEqual(result[0].outcome_group['id'], 2)
+            self.assertEqual(result[0].outcome_group['title'], "Global Test Outcome Group")
+
+            result = self.course_outcome_group.list_linked_outcomes()
+            self.assertIsInstance(result[0], OutcomeLink)
+            self.assertEqual(result[0].outcome_group['id'], 2)
+            self.assertEqual(result[0].outcome_group['title'], "Course Test Outcome Group")
+
+            self.assertEqual(len(warning_list), 1)
```

Comment:
Python 2 and Python 3 handle warnings differently. Python 3 doesn't consider duplicates of the same warning when catching and counting.

In this test's case, the DeprecationWarning is raised 3 times, but caught only once in Python 3.

In Python 2, a warning is counted each time it is caught. This means this line will raise `AssertionError: 3 != 1`, as 3 separate warnings are raised.